libparsec/crates/client/tests/unit/certif/add_device_certificate.rs

```diff
@@ -27,6 +27,27 @@ async fn ok(env: &TestbedEnv) {
     p_assert_matches!(switch, MaybeRedactedSwitch::NoSwitch);
 }
 
+#[parsec_test(testbed = "minimal")]
+async fn multiple(env: &TestbedEnv) {
```

Comment:
I think the name is not clear enough

The `ok` test ensure we can add all the certificates (i.e. the one we are interested into, and the ones it depends on) in an empty certif ops
The `multiple` test ensure we can add multiple certificates of the same type

What about:
- rename `ok` tests to `ok_from_scratch`
- add a call to `builder.certificates_storage_fetch_certificates()` so that the `multiple` test only insert the certificates we are interested into (and not the certificates they depend on)
- rename `multiple` tests to `ok_only_2_additional_certifs` (I'm okay with a better name ^^)

Reply:
It can be done in another PR to change the tests name, because this PR adds some tests (not all), so there are others which need to be changed.

I agree that the 2nd point is cleaner, but that doesn't bring more and the test would be longer

Reply:
> I agree that the 2nd point is cleaner,

What point are you talking about ?

Reply:
> add a call to builder.certificates_storage_fetch_certificates() so that the multiple test only insert the certificates we are interested into (and not the certificates they depend on)

Reply:
> I agree that the 2nd point is cleaner, but that doesn't bring more and the test would be longer

It is an important test though: we are testing the certificate ops is able to go fetch the requested data from it local storage.
If we add all certificates at once, we are in the case the certificates are only fetched from the cache.

So I really think we should do that to have a good coverage !

Reply:
What about testing without the cache then ?

Reply:
In which situation is it different ? Can we ensure that for every certificate: adding certificate is the same as fetching at a moment ?

Reply:
> What about testing with/without the cache then ?

That's precisely the point of using `builder.certificates_storage_fetch_certificates()`

Reply:
> In which situation is it different ?

Every time the cache is involved, there is two possible outcome: cache hit or cache miss. Hence both must be tested.

> Can we ensure that for every certificate: adding certificate is the same as fetching at a moment ?

That's the point of those tests.

Reply:
Done